app/scripts/modules/amazon/validation/applicationName.validator.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const angular = require('angular');
+
+module.exports = angular
+  .module('spinnaker.aws.validation.applicationName', [
+    require('../../core/application/modal/validation/applicationName.validator.js'),
+  ])
+  .factory('awsApplicationNameValidator', function () {
+
+    function validate(name) {
+      let warnings = [],
+          errors = [];
+      name = name || '';
+      if (name.indexOf('.') > -1 || name.indexOf('_') > -1) {
+        warnings.push(`If the application's name contains an underscore(_) or dot(.),
+          you will not be able to create a load balancer,
+          preventing it from being used as a front end service.`);
+      }
+
+      if (name.length > 20) {
+        if (name.length > 32) {
+          warnings.push(`You will not be able to create an Amazon load balancer for this application if the
+          application's name is 32 characters or longer (currently: ${name.length} characters)`);
+        } else {
+          let remaining = 30 - name.length;
```

Comment:
Good catch. Fixed to include a separate message when name is between 30 and 32 characters.